libraries/botbuilder-ai/tests/luisRecognizerTests.js

```diff
@@ -19,43 +20,191 @@ describe('LuisRecognizer', function () {
-        var context = { request: { text: 'I want ham and cheese sandwich!' } };
-        recognizer.recognize(context)
-            .then(res => {
+        var context = { request: { text: 'My name is Emad' } };
+        recognizer.recognize(context).then(res => {
                 assert(res);
-                assert(res.length == 1);
-                assert(res[0].name == 'sandwichorder');
-                assert(res[0].entities.length > 0);
-                assert(res[0].entities[0].type = 'meat');
-                assert(res[0].entities[0].value = 'ham');
+                assert(res.length === 1);
+                assert(res[0].text == 'My name is Emad');
+                assert(Object.keys(res[0].intents).length > 1);
+                assert(res[0].intents.SpecifyName);
+                assert(res[0].intents.SpecifyName > 0 && res[0].intents.SpecifyName <= 1);
+                assert(res[0].entities);
+                assert(res[0].entities.Name);
+                assert(res[0].entities.Name[0] === 'emad');
+                assert(res[0].$instance);
+                assert(res[0].$instance.entities);
+                assert(res[0].$instance.entities.Name);
+                assert(res[0].$instance.entities.Name[0].$startIndex === 11);
+                assert(res[0].$instance.entities.Name[0].$endIndex === 14);
+                assert(res[0].$instance.entities.Name[0].$score > 0 && res[0].$instance.entities.Name[0].$score <= 1);
                 done();
             });
     });
 
-    it('static recognize should return top intent', function (done) {
-        ai.LuisRecognizer.recognize("hello", { appId: luisAppId, subscriptionKey: subscriptionKey })
-            .then(res => {
+    it('should return multiple intents and prebuilt entities with a single value', function(done){
+        var recognizer = new ai.LuisRecognizer({ appId: luisAppId, subscriptionKey: subscriptionKey });
+        var context = { request: { text: 'Please deliver February 2nd 2001' } };
+        recognizer.recognize(context).then(res => {
                 assert(res);
-                assert(res.name != '');
-                assert(res.entities.length == 0);
+                assert(res.length === 1);
+                assert(res[0].text == 'Please deliver February 2nd 2001');
+                assert(res[0].intents);
+                assert(res[0].intents.Delivery);
+                assert(res[0].intents.Delivery > 0 && res[0].intents.Delivery <= 1);
+                assert(res[0].entities);
+                assert(res[0].entities['builtin.number']);
+                assert(res[0].entities['builtin.number'][0] === 2001);
+                assert(res[0].entities['builtin.datetimeV2.date']);
+                assert(res[0].entities['builtin.datetimeV2.date'][0] === '2001-02-02');
+                assert(res[0].$instance);
+                assert(res[0].$instance.entities);
+                assert(res[0].$instance.entities['builtin.number']);
```

Comment:
Were you going to map from . to _ or :?

Reply:
I still need to give this more thought. : is a no go since it's not valid JS. _ is reasonable but it looks ugly. I'll use _ for now but I'm not very satisfied with it, might change it later.